File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/29_compito_java_2019_04_11b/Visitatore.java
import java.util.*;

public class Visitatore {

	private int codice;
	private String nomeCognome;
	private List<Servizio> servizi;

	public Visitatore(int codice, String nomeCognome) {
		this.codice = codice;
		this.nomeCognome = nomeCognome;
		this.servizi = new LinkedList<Servizio>();
	}

	public void addServizio(Impiegato dip, double ore) {
		servizi.add(new Servizio(dip, ore));
	}

	public double getTot() {
		double tot = 0;
		for (Servizio s : servizi)
			tot = tot + s.getValore();
		return tot;

	}

	public String toString() {
		return codice + " " + nomeCognome + " " + servizi;
	}
}

File: II - Linguaggi di programmazione e laboratorio/Esame/Python/29b_compito_java_20190411/Visitatore.py
class Visitatore:
    def __init__(self, codice_visitatore, nome_visitatore):
        self._codice_visitatore = codice_visitatore
        self._nome_visitatore = nome_visitatore
        self._servizi = []

    def add_servizio(self, s):
        self._servizi.append(s)

    def __str__(self):
        return str(self._codice_visitatore) + " " + self._nome_visitatore + " " + str([str(s) for s in self._servizi])
